escape_roomba/conftest.py

```diff
@@ -0,0 +1,129 @@
+"""Shared configuration for unit tests."""
```

Comment:
What I've seen most commonly in Python projects is tests being placed in a `<repo_root>/tests/` directory, which then becomes a clear place to put a `tests/config.py` file (and you don't have to have a file lying around called "conftest").

Here is a sample Python project that demostrates this setup: https://github.com/navdeep-G/samplemod/tree/master/tests

Since you are using Poetry, and thus the `escape_roomba` package is "installed" in the environment, you wouldn't even need the customary `context.py` file to wrangle the imports.

FWIW, Pytest themselves also seem to assume that as a default setup (although of course they support other modes), as in these examples: https://docs.pytest.org/en/stable/example/pythoncollection.html

Reply:
"conftest.py" is a magic name for pytest, and seems to be the standard/recommended place to put shared fixtures:
https://docs.pytest.org/en/stable/fixture.html#conftest-py-sharing-fixture-functions

As for layout, pytest offers this advice:
https://docs.pytest.org/en/stable/goodpractices.html#choosing-a-test-layout-import-rules
https://docs.pytest.org/en/stable/pythonpath.html#import-modes

You're right that what I'm doing (intermingling tests and modules, Google-style) is not any of the recommended things. Moved into a separate top-level tests/ directory.

Reply:
Magic names! What Python is this?! 🙀

Reply:
The business they do with injecting values based on _parameter name_ is also pretty un-Pythonically magical!